src/plugins/discover/public/application/main/hooks/use_test_based_query_language.test.tsx

```diff
@@ -270,7 +268,7 @@ describe('useTextBasedQueryLanguage', () => {
       ],
       query: { sql: 'SELECT field1 from the-data-view-title' },
     });
-    await waitFor(() => expect(replaceUrlState).toHaveBeenCalledTimes(1));
+    await waitFor(() => expect(replaceUrlState).toHaveBeenCalledTimes(2));
```

Comment:
@kertal can you take this? @davismcphee I asked Matthias help on this change as I couldn't find easy making these changes due to the complexity of the discover state and the keep the url state in sync. So the majority of the changes in Discover were made by Matthias

Reply:
Sure thing! Because we have 2 queries here, and now since we have only ad-hoc data view with the first query, the index of the ad-hoc one has to replaced in URL. Secondly, the fields are changed, so those also have to be replaced in the URL

Reply:
From my understanding on the first fetch it adds the index to the state and on the second the columns. If you have any idea how we simplify it I am open but I have tried some combinations and they create bugs on the logic.

Reply:
I don't think there's a way to get just 1 update, when there are 2 different queries, so this is fine 👍

Reply:
Makes sense, thanks!